lib/Translations.php

```diff
@@ -92,21 +92,18 @@ public function make_pots() {
 	}
 
 	/**
-	 * Wrap jsons with a layer with the key 'locale_data'.
+	 * Apply transformer given the format for the data.
 	 *
-	 * @since  0.0.0
-	 * @param  array $jsons
-	 * @return array
+	 * @since 0.0.0
+	 * @return mixed
 	 */
-	private function wrap_jsons( array $jsons ) : array {
-		foreach ( $jsons as $key => $json ) {
-			$jsons[ $key ] = json_encode(
-				[
-					'locale_data' => json_decode( $json, true ),
-				],
-				JSON_PRETTY_PRINT
-			);
+	private function apply_transformer( $data, Project $config ) {
+		//TODO: receive outside classes.
+		$transformer_class = __NAMESPACE__ . '\\Transformers\\' . ucfirst( $config->get_format() );
+		if ( class_exists( $transformer_class ) ) {
+			$transfomer = new $transformer_class();
+			return $transfomer->transform( $data, $config );
```

Comment:
O `$transformer_class` que tens agora poderá ser a base, mas seria bom poder receber uma classe de fora para ser mais extensível. Futuro pull request ;)

O resto da lógica para remover complexidade e aumentar o que é testado.

```suggestion
		if ( ! class_exists( $transformer_class ) ) {
			return $data;
		}
		
		$transfomer = new $transformer_class();
		if ( ! $transformer instanceof Transformer ) {
			return $data;
		}
		
		return $transfomer->transform( $data, $config );
```